<a href="https://colab.research.google.com/github/krisbarbier/Project-2/blob/main/Kris_B_Project2_Part1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Project 2: Proposal

- Student: Kris Barbier

##Dataset 1: Metabolic Syndrome Prediction

In [1]:
#Import pandas
import pandas as pd


In [2]:
#Mount Google Drive
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
#Load in dataset and preview first 5 rows
fpath = '/content/drive/MyDrive/CodingDojo/02-MachineLearning/Metabolic  Syndrome.csv'

df = pd.read_csv(fpath)

df.head()

seqn  Age     Sex  Marital  Income   Race  WaistCirc   BMI  Albuminuria  \
0  62161   22    Male   Single  8200.0  White       81.0  23.3            0   
1  62164   44  Female  Married  4500.0  White       80.1  23.2            0   
2  62169   21    Male   Single   800.0  Asian       69.6  20.1            0   
3  62172   43  Female   Single  2000.0  Black      120.4  33.3            0   
4  62177   51    Male  Married     NaN  Asian       81.1  20.1            0   

   UrAlbCr  UricAcid  BloodGlucose  HDL  Triglycerides MetabolicSyndrome  
0     3.88       4.9            92   41             84         No MetSyn  
1     8.55       4.5            82   28             56         No MetSyn  
2     5.07       5.4           107   43             78         No MetSyn  
3     5.22       5.0           104   73            141         No MetSyn  
4     8.13       5.0            95   43            126         No MetSyn

In [4]:
#Check shape
print(f'There are {df.shape[0]} rows and {df.shape[1]} columns.')

There are 2401 rows and 15 columns.


First choice: dataset 1

1. Source of data

  - This data comes from data.world: https://data.world/informatics-edu/metabolic-syndrome-prediction

2. Brief description of data

  - This data is used to predict is a person will be diagnosed with metabolic syndrome, which is a combination of factors that lead to the onset of cardiovascular disease and type 2 diabetes. The data set includes features that directly relate to the presence of metabolic syndrome, which include measures of obesity (waist cirumference and bmi), triglyceride levels, cholesterol levels, and blood glucose levels. While these are the main values to predict metabolic syndrome, the data also includes other factors, such as age, race, income level, and measures of uric acid. I believe that this dataset will provide some interesting insights into metabolic syndrome across many different demographics.

3. What is the target?

  - The target is the 'MetabolicSyndrome' feature.

4. What does one row represent? (A person? A business? An event? A product?)

  - In this data, one row represents one person (patient).

5. Is this a classification or regression problem?

  - This will be a classification problem.

6. How many features does the data have?

  - The data has 15 features.

7. How many rows are in the dataset?

  - There are 2401 rows in the dataset.

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

  - The main challenges that I foresee will be related to exploring and modeling the data. When it comes to exploring the data, I think it could become hard to keep a narrow focus on the purpose of the data analysis. While the predicted target will always be whether the patient will develop metabolic syndrome or not, there are many different ways to explore this with the other features present, so it will be wise to stay focused on the purpose of the task at hand. In terms of modeling, it will be challenging to find the right models to use to get the best results for this task.

##Dataset 2: Student Performance in Exams

In [5]:
#Load in data and preview first 5 rows
filepath = '/content/drive/MyDrive/CodingDojo/02-MachineLearning/StudentsPerformance.csv'

df2 = pd.read_csv(filepath)

df2.head()

gender race/ethnicity parental level of education         lunch  \
0  female        group B           bachelor's degree      standard   
1  female        group C                some college      standard   
2  female        group B             master's degree      standard   
3    male        group A          associate's degree  free/reduced   
4    male        group C                some college      standard   

  test preparation course  math score  reading score  writing score  
0                    none          72             72             74  
1               completed          69             90             88  
2                    none          90             95             93  
3                    none          47             57             44  
4                    none          76             78             75

In [6]:
#Check shape
print(f'There are {df2.shape[0]} rows and {df2.shape[1]} columns.')

There are 1000 rows and 8 columns.


Second choice: Dataset 2

1. Source of data

  - This data comes from: http://roycekimmons.com/tools/generated_data/exams and is referenced on Kaggle.

2. Brief description of data

  - This data is used to predict how well students will score on different exams based on different things incuding gender, race, parental level of education, and whether or not they prepared for the exams. Coming from the world of teaching and education, and having looked at many different types of student performance data on assessments, this dataset really intrigues me.

3. What is the target?

  - The target for this dataset could be one of three different targets: math score, reading score, or writing score. I will have to decide which score is the most important one to classify for this project, or maybe even try to combine the three into a new category that represents all scores together.

4. What does one row represent? (A person? A business? An event? A product?)

  - In this data, one row represents one person (student).

5. Is this a classification or regression problem?

  - This will be a regression problem if the targets are used as is. If a new feature is developed to combine the three scores, it could remain a regression problem (if all three scores are added together or averaged) or become a classification problem (if the combination becomes a feature using pass/fail or other related classes).

6. How many features does the data have?

  - The data has 8 features.

7. How many rows are in the dataset?

  - There are 1000 rows in the dataset.

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

  - The main challenge that I think will occur with this dataset is trying to figure out how to set the target, and then to proceed from there with regression or classification. Cleaning should be relatively easy for this dataset, as there are no missing values to begin with.